drivers/espi/espi_saf_mchp_xec_v2.c

```diff
@@ -0,0 +1,1164 @@
+}
+
+static void espi_saf_err_isr(const struct device *dev)
+{
+	const struct espi_saf_xec_config * const xcfg = dev->config;
+	struct espi_saf_xec_data *data = dev->data;
+	struct mchp_espi_saf * const regs = xcfg->saf_base;
+	const struct espi_xec_irq_info *safirq = &xcfg->irq_info_list[1];
+	uint32_t mon_status = regs->SAF_ESPI_MON_STATUS;
+	struct espi_event evt = { .evt_type = ESPI_BUS_PERIPHERAL_NOTIFICATION,
+				  .evt_details = BIT(7),
+				  .evt_data = mon_status };
+
+	regs->SAF_ESPI_MON_STATUS = mon_status;
+	mchp_xec_ecia_girq_src_clr(safirq->gid, safirq->gpos);
+
+	data->hwstatus = mon_status;
+	espi_send_callbacks(&data->callbacks, dev, evt);
+}
+
+static int espi_saf_xec_init(const struct device *dev);
+
+static const struct espi_saf_driver_api espi_saf_xec_driver_api = {
+	.config = espi_saf_xec_configuration,
+	.set_protection_regions = espi_saf_xec_set_pr,
+	.activate = espi_saf_xec_activate,
+	.get_channel_status = espi_saf_xec_channel_ready,
+	.flash_read = saf_xec_flash_read,
+	.flash_write = saf_xec_flash_write,
+	.flash_erase = saf_xec_flash_erase,
+	.manage_callback = espi_saf_xec_manage_callback,
+};
+
+static struct espi_saf_xec_data espisaf_xec_data;
+
+/* n = node-id, p = property, i = index */
+#define XEC_SAF_IRQ_INFO(n, p, i)					    \
+	{								    \
+		.gid = MCHP_XEC_ECIA_GIRQ(DT_PROP_BY_IDX(n, p, i)),	    \
+		.gpos = MCHP_XEC_ECIA_GIRQ_POS(DT_PROP_BY_IDX(n, p, i)),    \
+		.anid = MCHP_XEC_ECIA_NVIC_AGGR(DT_PROP_BY_IDX(n, p, i)),   \
+		.dnid = MCHP_XEC_ECIA_NVIC_DIRECT(DT_PROP_BY_IDX(n, p, i)), \
+	},
+
+static const struct espi_xec_irq_info espi_saf_xec_irq_info_0[] = {
+	DT_FOREACH_PROP_ELEM(DT_NODELABEL(espi_saf0), girqs, XEC_SAF_IRQ_INFO)
+};
+
+static const struct espi_saf_xec_config espisaf_xec_config = {
+	.saf_base = (struct mchp_espi_saf * const)(DT_INST_REG_ADDR_BY_IDX(0, 0)),
+	.qmspi_base = (struct qmspi_regs * const)(DT_INST_REG_ADDR_BY_IDX(0, 1)),
+	.saf_comm_base = (struct mchp_espi_saf_comm * const)(DT_INST_REG_ADDR_BY_IDX(0, 2)),
+	.iom_base = (struct espi_iom_regs * const)(DT_REG_ADDR_BY_NAME(DT_INST_PARENT(0), io)),
+	.poll_timeout = DT_INST_PROP_OR(inst, poll_timeout,
+					MCHP_SAF_FLASH_POLL_TIMEOUT),
+	.consec_rd_timeout = DT_INST_PROP_OR(
+		inst, consec_rd_timeout, MCHP_SAF_FLASH_CONSEC_READ_TIMEOUT),
+	.sus_chk_delay = DT_INST_PROP_OR(inst, sus_chk_delay,
+					 MCHP_SAF_FLASH_SUS_CHK_DELAY),
+	.sus_rsm_interval = DT_INST_PROP_OR(inst, sus_rsm_interval,
+					    MCHP_SAF_FLASH_SUS_RSM_INTERVAL),
+	.poll_interval = DT_INST_PROP_OR(inst, poll_interval,
+					 MCHP_SAF_FLASH_POLL_INTERVAL),
+	.pcr_idx = DT_INST_PROP_BY_IDX(0, pcrs, 0),
+	.pcr_pos = DT_INST_PROP_BY_IDX(0, pcrs, 1),
+	.irq_info_size = ARRAY_SIZE(espi_saf_xec_irq_info_0),
+	.irq_info_list = espi_saf_xec_irq_info_0,
+};
+
+DEVICE_DT_INST_DEFINE(0, &espi_saf_xec_init, NULL,
+		      &espisaf_xec_data, &espisaf_xec_config, POST_KERNEL,
+		      CONFIG_ESPI_SAF_INIT_PRIORITY, &espi_saf_xec_driver_api);
```

Comment:
Please use `DT_INST_FOREACH_STATUS_OKAY`

Reply:
It is a single espi saf endpoint. We don't envision multiple saf (slave attached flash) instance.

Reply:
updated